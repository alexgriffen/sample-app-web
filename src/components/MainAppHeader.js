import React, { Component } from 'react';
import * as PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import { withStyles } from '@material-ui/core';
import { colors } from '../utils/Colors';
import headerLogoPng from '../assets/img/header-logo.png';
import headerLogoSvg from '../assets/svg/logo@3x.svg';
import MenuIcon from './Icons/MenuIcon';
import CartIcon from './Icons/CartIcon';

const styles = theme => ({
    grow: {
      flexGrow: 1,
    },
    root: {
      backgroundColor: colors.white,
      boxShadow: 'none',
      height: 80,
      [ theme.breakpoints.down('md') ]: {
        height: 60,
      },
    },
    toolBar: {
      flex: 1,
      justifyContent: 'space-between',
    },
    headerLogo: {
      height: '40px',
      width: '259px',
      [ theme.breakpoints.down('md') ]: {
        minHeight: 20,
        maxWidth: 190,
      },
    },
  }
);
const { object } = PropTypes;

class MainAppHeader extends Component {
  constructor(props) {
    super(props);

    this.handleMenu = this.handleMenu.bind(this);
    this.openCart = this.openCart.bind(this);
  }

  static propTypes = {
    classes: object.isRequired,
  };

  handleMenu() {
    // Open the drawer
  }

  openCart() {
    // Open the drawer
  }

  render() {
    const { classes } = this.props;

    return (
      <div className={ classes.grow }>
        <AppBar className={ classes.root }>
          <Toolbar className={ classes.toolBar }>
            <MenuIcon
              fallBackClasses="bm-burger-button"
              onClick={ this.handleMenu() }
              testID="open-menu-icon"
            />
            <img src={ headerLogoPng } srcSet={ headerLogoSvg } className={ classes.headerLogo }/>
            <CartIcon
              fallBackClasses="shopping_cart_link"
              onClick={ this.openCart() }
              testID="cart-icon"
              cartContent={1}
            />
          </Toolbar>
        </AppBar>
      </div>
    );
  }
}

export default withStyles(styles)(MainAppHeader);
