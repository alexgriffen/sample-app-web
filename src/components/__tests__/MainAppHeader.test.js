import React from 'react';
import { shallow } from 'enzyme/build';
import MainAppHeader from '../MainAppHeader';

describe('Main App Header', () => {
  it('should render the component correctly', () => {
    const component = shallow(<MainAppHeader />).dive();

    expect(component).toMatchSnapshot();
  });
});
