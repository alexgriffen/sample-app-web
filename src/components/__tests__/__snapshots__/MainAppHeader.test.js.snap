// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main App Header should render the component correctly 1`] = `
<WithStyles(AppBar)
  className="MainAppHeader-root-1"
>
  <WithStyles(Toolbar)
    className="MainAppHeader-toolBar-2"
  >
    <WithStyles(MenuIcon)
      fallBackClasses="bm-burger-button"
      testID="open-menu-icon"
    />
    <img
      className="MainAppHeader-headerLogo-3"
      src="test-file-stub"
      srcSet="test-file-stub"
    />
    <WithStyles(CartIcon)
      cartContent={1}
      fallBackClasses="shopping_cart_link"
      testID="cart-icon"
    />
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
